internal/provider/gce/environ_firewall.go

```diff
@@ -62,18 +78,32 @@ func firewallSpec(name, target string, sourceCIDRs []string, ports protocolPorts
 // If a rule matching a set of source ranges doesn't
 // already exist, it will be created - the name will be made unique
 // using a random suffix.
-
 func (env *environ) OpenPorts(ctx context.ProviderCallContext, target string, rules firewall.IngressRules) error {
-	err := env.openPorts(ctx, target, rules)
-	return google.HandleCredentialError(errors.Trace(err), ctx)
+	// Separate IPv4 and IPv6 rules to avoid mixed CIDRs in a single firewall, which is disallowed by GCE.
+	// Ensure default SourceCIDRs are set for any rule with empty SourceCIDRs before separating rules.
+	ensureSourceCIDRs(&rules)
+	ipv4Rules := rules.RemoveCIDRsMatchingAddressType(network.IPv6Address)
```

Comment:
The firewaller worker already strips out ipv6 cidrs as needed, so we don't need to do it here.'

All of the code here can be deleted - this OpenPorts method for instances is already correctly used by the firewaller.